run-go-mod-tidy.sh

```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+if go mod tidy -v 2>&1 | grep 'unused'; then
```

Comment:
There are other potential warnings here, like `go: updates to go.mod needed, but contents have changed`.

Should this error on any output?

Reply:
I never had this warning, how did you achieved this?

Reply:
I think I added a new (unused) dependency to `go.mod` and ran `go mod tidy` before any running any other go command.

Reply:
@dnephin I updated my script to update `go.mod` `go.sum` files (update needed packages, remove useless ones), catch the "need update" error you find to ask manually modifications. If no errors are catch then it re-add the corrected `go.mod` and `go.sum` file to the commit.

Reply:
Finally there is no need to `exit 1` when `unused` dependencies are found because the command correct the files directly, so we just need to re-add it to the commit.

I didn't find other potential warnings/errors when checking the command docs or during months of command use.

cf: https://tip.golang.org/cmd/go/#hdr-Add_missing_and_remove_unused_modules